frontend/src/pages/journal/journal.tsx

```diff
@@ -1,16 +1,47 @@
 import { BackButton } from '#libs/components/components.js';
+import { AppRoute } from '#libs/enums/enums.js';
 import { getValidClassNames } from '#libs/helpers/helpers.js';
-import { useCallback, useSidebarState } from '#libs/hooks/hooks.js';
+import {
+  useAppDispatch,
+  useAppSelector,
+  useCallback,
+  useEffect,
+  useNavigate,
+  useParams,
+  useSidebarState,
+} from '#libs/hooks/hooks.js';
+import { actions as journalActions } from '#slices/journal/journal.js';
 
 import { JournalSidebar } from './components/journal-sidebar/journal-sidebar.js';
+import { Note } from './components/note/note.js';
 import styles from './styles.module.scss';
 
 const Journal: React.FC = () => {
+  const { id } = useParams();
+  const navigate = useNavigate();
+  const dispatch = useAppDispatch();
   const { isSidebarShown, setIsSidebarShown } = useSidebarState();
 
+  const { selectedJournalEntry } = useAppSelector(({ journal }) => {
+    return {
+      selectedJournalEntry: journal.selectedJournalEntry,
+    };
+  });
+
+  const hasNoteId = Boolean(id) && selectedJournalEntry;
```

Comment:
let's rename it to this to make more logical
```suggestion
  const hasSelectedNote = Boolean(id) && selectedJournalEntry;
```

Reply:
Done